spark-connector/spark-common/src/test/java/com/datastrato/gravitino/spark/connector/integration/test/sql/SparkQueryRunner.java

```diff
@@ -0,0 +1,338 @@
+    String hiveMetastoreUri =
+        String.format(
+            "thrift://%s:%d",
+            containerSuite.getHiveContainer().getContainerIpAddress(),
+            HiveContainer.HIVE_METASTORE_PORT);
+    this.warehouse =
+        String.format(
+            "hdfs://%s:%d/user/hive/warehouse",
+            containerSuite.getHiveContainer().getContainerIpAddress(),
+            HiveContainer.HDFS_DEFAULTFS_PORT);
+
+    // start Gravitino server
+    AbstractIT.startIntegrationTest();
+    int gravitinoPort = AbstractIT.getGravitinoServerPort();
+    this.gravitinoUri = String.format("http://127.0.0.1:%d", gravitinoPort);
+
+    // init metalake and catalog
+    GravitinoAdminClient client = AbstractIT.getGravitinoClient();
+    client.createMetalake(metalakeName, "", Collections.emptyMap());
+    GravitinoMetalake metalake = client.loadMetalake(metalakeName);
+    metalake.createCatalog(
+        HIVE_CATALOG_NAME,
+        Catalog.Type.RELATIONAL,
+        "hive",
+        "",
+        getHiveCatalogConfigs(hiveMetastoreUri));
+    metalake.createCatalog(
+        ICEBERG_CATALOG_NAME,
+        Catalog.Type.RELATIONAL,
+        "lakehouse-iceberg",
+        "",
+        getIcebergCatalogConfigs(hiveMetastoreUri));
+
+    client.close();
+  }
+
+  private Map<String, String> getHiveCatalogConfigs(String hiveMetastoreUri) {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    catalogProperties.put(GravitinoSparkConfig.GRAVITINO_HIVE_METASTORE_URI, hiveMetastoreUri);
+    return catalogProperties;
+  }
+
+  protected Map<String, String> getIcebergCatalogConfigs(String hiveMetastoreUri) {
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_BACKEND,
+        IcebergPropertiesConstants.ICEBERG_CATALOG_BACKEND_HIVE);
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_WAREHOUSE, warehouse);
+    catalogProperties.put(
+        IcebergPropertiesConstants.GRAVITINO_ICEBERG_CATALOG_URI, hiveMetastoreUri);
+    return catalogProperties;
+  }
+
+  private void closeGravitinoEnv() throws Exception {
+    AbstractIT.stopIntegrationTest();
+  }
+
+  private void writeQueryOutput(Path outputFile, List<QueryOutput> queryOutputs)
+      throws IOException {
+    String goldenOutput =
+        "-- Automatically generated by Gravitino Spark SQL test\n"
+            + queryOutputs.stream().map(QueryOutput::toString).collect(Collectors.joining("\n\n\n"))
+            + "\n";
+    stringToFile(outputFile, goldenOutput);
+  }
+
+  private List<String> getQueriesFromFile(Path file) throws IOException {
+    String input = fileToString(file);
+    Pair<List<String>, List<String>> pair = splitCommentsAndCodes(input);
+    List<String> queries = pair.getRight();
+    queries = splitWithSemicolon(queries);
+    queries = cleanAndFilterQueries(queries);
+    return queries;
+  }
```

Comment:
Is the test system only used in  the local gravitino and spark environment.

Reply:
In this PR yes, we could extend it in another PR